xscen/catalog.py

```diff
@@ -347,6 +347,115 @@ def exists_in_cat(self, **columns):
             logger.info(f"An entry exists for: {columns}")
         return exists
 
+    def to_dataset(
+        self,
+        concat_on: Optional[Union[List[str], str]] = None,
+        ensemble_on: Optional[Union[List[str], str]] = None,
+        calendar: Optional[str] = "standard",
+        **kwargs,
+    ) -> xr.Dataset:
+        """
+        Open the catalog's entries into a single dataset.
+
+        Same as :py:meth:`~intake_esm.core.esm_datastore.to_dask`, but with additionnal control over the aggregations.
+        Ensemble preprocessing logic is taken from :py:func:`xclim.ensembles.create_ensemble`.
+        When `ensemble_on` is given, the function ensures all entries have the correct time coordinate according to `xrfreq`.
+        If either argument is given, the "id" is reconstructed by removing mentions of aggregated columns.
+        This will override any "custom" id, ones not unstackable with :py:func:`~xscen.catalog.unstack_id`.
+
+        Parameters
+        ----------
+        concat_on : list of strings or str, optional
+          A list of catalog columns to concat the datasets over. Each will become a new dimension with the column values as coordinates.
+          Xarray concatenation rules apply and can be acted upon through `xarray_combine_by_coords_kwargs`.
+        ensemble_on : list of strings or str, optional
```

Comment:
```suggestion
        create_ensemble_on : list of strings or str, optional
```
Maybe? To match the xclim name (since this is essentially what this does).

Reply:
Also, a more natural logic for this argument (to me at least) would be something like : `create_ensemble_on: ['member']` (where we then remove the member from `id` and create the ensemble), instead of having to list all the columns except the one that actually matters.

Reply:
Personnally, I like the "put the columns that you want" logic better.

Reply:
Not sure I understand, @RondeauG , aren't you describing what this function does?

The difference with `concat_on` is that `ensemble_on` will merge the given column values into a new id-like column. Then it adds an aggregation over that new column and removes the given columnd from the dataset id, so as to remove them from the `groupby_attrs` logic. 

I have no hard feelings against `create_ensemble_on`.